trin-core/src/types/block_body.rs

```diff
@@ -57,6 +47,73 @@ impl Uncles {
     }
 }
 
+impl TryFrom<BlockBodyHelper> for BlockBody {
+    type Error = ssz::DecodeError;
+
+    fn try_from(helper: BlockBodyHelper) -> Result<Self, Self::Error> {
+        let txs: Vec<Transaction> = helper
+            .rlp_txs
+            .iter()
+            .map(|bytes| rlp::decode(bytes).unwrap())
+            .collect();
+        let uncles: Vec<Header> = helper
+            .rlp_uncles
+            .iter()
+            .map(|bytes| rlp::decode(bytes).unwrap())
+            .collect();
+
+        Ok(Self { txs, uncles })
+    }
+}
+
+impl ssz::Encode for BlockBody {
+    fn is_ssz_fixed_len() -> bool {
+        false
+    }
+
+    fn ssz_append(&self, buf: &mut Vec<u8>) {
+        BlockBodyHelper::from(self).ssz_append(buf);
+    }
+
+    fn ssz_bytes_len(&self) -> usize {
+        self.as_ssz_bytes().len()
+    }
+}
+
+impl ssz::Decode for BlockBody {
+    fn is_ssz_fixed_len() -> bool {
+        false
+    }
+
+    fn from_ssz_bytes(bytes: &[u8]) -> Result<Self, ssz::DecodeError> {
+        let helper = BlockBodyHelper::from_ssz_bytes(bytes)?.try_into()?;
+        Ok(helper)
+    }
+}
+
+#[derive(Debug, Decode, Encode)]
+struct BlockBodyHelper {
```

Comment:
Whenever I write a ...`Helper` name, I'm pretty sure that I'm not done naming it.

But yeah, clearly a more challenging name. `EncodedBlockBodyParts?` ick. `EncodedBlockElements` maybe?